Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -130,8 +130,66 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
                         return deferMaybe(result)
                     }
                 }
-
                 return deferMaybe(result)
+            } >>== { result in
+                guard case .completed = result else {
```

Comment:
I don't think this is true, [we also return .completed when there's no new data](https://github.com/mozilla-mobile/firefox-ios/blob/ec471414cbfaa31a4c27cbec855c82d5fa50d5e6/Sync/Synchronizers/Bookmarks/BookmarksDownloader.swift#L147)